### ISSUE TITLE
Auto-update working?

### ISSUE THREAD

**Issue by alansill** [spack/packages.spack.io#3]
Just noticed that the last time the packages have been refreshed by the auto-update was July 12. Is the automated GitHub action working? Or am I misreading the structure of how the package list is updated?

**Comment by vsoch** [comment]
Workflows are automatically disabled after a certain number of days:

![image](https://user-images.githubusercontent.com/814322/200143285-c9a0b891-81b1-49fb-a77f-929badf104e9.png)

@tgamblin this happened before and I sounded the alarm - I guess you didn't see a notification this time either? I'm not even on this repository with any permissions so it's sort of out of my hands. The workflow needs to be -re-enbabled by someone ith permissions, and then someone needs to make sure they are notified when the email goes out (based on my other repos, the emails are sent out and then I know to click the button to re-enable).

Sorry @alansill !

**Comment by alansill** [comment]
I could help with this if you want. (your call @tgamblin !)

If I wanted to add another seearch field to search for text within the description, leaving the other elements of the display intact (i.e., the search just like the existing one for the package names would just show the cars for the names that have matches, but this field would search within the description text), what would I do?

**Comment by vsoch** [comment]
Short answer: JavaScript!

Long answer: You'd need to populate that data somewhere on the page to start, and then associate each description entry with a particular name. You'd probably want to parse all the package descriptions into a new description endpoint first (e.g., data/descriptions.json) and then load into the page and associate with the existing elements as tokens. I'd check out tagify.js - it's what I use for https://software.llnl.gov/radiuss/projects/. It's actually exactly this same strategy, but that one is rendered from jekyll logic and this site is pure json -> javascript.

**Comment by vsoch** [comment]
@tgamblin please re-enable the workflow!

**Comment by tgamblin** [comment]
Gah -- sorry @alansill -- I didn't get a chance to trigger this but just did.  I'm going to put a task in place to keep re-enabling the workflow

**Comment by tgamblin** [comment]
For now I added both @vsoch and @alansill to the repo, as well -- so both of you should be able to re-trigger the workflow here: https://github.com/spack/packages.spack.io/actions/workflows/update-ui.yaml

**Comment by vsoch** [comment]
oh good spack broke something again... :sob: :laughing: 

```
Run export PATH=/opt/spackydoo/bin:$PATH
Traceback (most recent call last):
Parsing testdfsio, 0 of 6748
  File "/usr/lib/python3.8/code.py", line 90, in runcode
    exec(code, self.locals)
  File "generate_packages.py", line 327, in <module>
    main()
  File "generate_packages.py", line 59, in main
    pkg = spack.spec.Spec(package).package
  File "/opt/spackydoo/lib/spack/spack/spec.py", line 1637, in package
    assert self.concrete, "Spec.package can only be called on concrete specs"
AssertionError: Spec.package can only be called on concrete specs
```

**Comment by tgamblin** [comment]
That's intentional -- we don't want people to try to install abstract specs.  The way to get the data is through the package class (which doesn't require instantiating a package)  -- we just finally started enforcing that.  I'll fix it

**Comment by tgamblin** [comment]
Ok this is fixed (and simplified a bit in 7ef2c733b30f635209d29b2bd8a55ca60cf25b8a.  I will keep working on it and get automation set up.

**Comment by tgamblin** [comment]
manually ran the workflow and it's now updated.
